pinot-query-runtime/src/test/java/org/apache/pinot/query/runtime/QueryRunnerTest.java

```diff
@@ -151,6 +151,8 @@ private Object[][] provideTestSql() {
         new Object[]{"SELECT * FROM b ORDER BY col1, col2 DESC LIMIT 3"},
         new Object[]{"SELECT * FROM a ORDER BY col1, ts LIMIT 10"},
         new Object[]{"SELECT * FROM a ORDER BY col1 LIMIT 20"},
+        new Object[]{"SELECT * FROM a ORDER BY col1, ts LIMIT 1, 2"},
+        new Object[]{"SELECT * FROM a ORDER BY col1, ts LIMIT 2 OFFSET 1"},
```

Comment:
FWIW - We don't support pagination (OFFSET / FETCH based syntax) even in the existing engine. Many times, pinot users have thought we support pagination (because syntax is supported) and the solution brute force for selection queries and does not work for other queries. So, probably we should not worry about adding this test.

Reply:
thanks for the observation @siddharthteotia - that makes me think that perhaps we should be doing the opposite and actually fail hard on queries where it isn't supported?

cc @walterddr - since I was actively working on that for multistage